### ISSUE TITLE
getAmplifyDataClientConfig() fails to check Amplify data variables when API name is defined

### ISSUE THREAD

**Issue by lengieng** [aws-amplify/amplify-backend#2320]
### Environment information

```plain text
System:
  OS: Windows 11 10.0.26100
  CPU: (12) x64 13th Gen Intel(R) Core(TM) i5-1335U
  Memory: 2.08 GB / 15.72 GB
Binaries:
  Node: 22.2.0 - C:\Program Files\nodejs\node.EXE
  Yarn: 1.22.22 - ~\AppData\Roaming\npm\yarn.CMD
  npm: 10.9.1 - C:\Program Files\nodejs\npm.CMD
  pnpm: undefined - undefined
NPM Packages:
  @aws-amplify/auth-construct: 1.5.1
  @aws-amplify/backend: 1.9.0
  @aws-amplify/backend-auth: 1.4.2
  @aws-amplify/backend-cli: 1.4.3
  @aws-amplify/backend-data: 1.2.2
  @aws-amplify/backend-deployer: 1.1.11
  @aws-amplify/backend-function: 1.9.0
  @aws-amplify/backend-output-schemas: 1.4.0
  @aws-amplify/backend-output-storage: 1.1.4
  @aws-amplify/backend-secret: 1.1.5
  @aws-amplify/backend-storage: 1.2.4
  @aws-amplify/cli-core: 1.2.1
  @aws-amplify/client-config: 1.5.3
  @aws-amplify/deployed-backend-client: 1.4.2
  @aws-amplify/form-generator: 1.0.3
  @aws-amplify/model-generator: 1.0.9
  @aws-amplify/platform-core: 1.3.0
  @aws-amplify/plugin-types: 1.6.0
  @aws-amplify/sandbox: 1.2.7
  @aws-amplify/schema-generator: 1.2.6
  aws-amplify: 6.10.2
  aws-cdk: 2.172.0
  aws-cdk-lib: 2.172.0
  typescript: 5.6.2
No AWS environment variables
No CDK environment variables
```

### Describe the bug

The doc shows an example of how to use `defineAuth` and `defineFunction` to create [Cognito post confirmation Lambda trigger](https://docs.amplify.aws/react/build-a-backend/functions/examples/create-user-profile-record/). Another related section is [grant Lambda function access to API](https://docs.amplify.aws/react/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/#access-the-api-using-aws-amplify).

When name is given in the `defineData()`, the generated `env` that start with the `AMPLIFY_DATA` prefix are replaced with the given name.
```
export const data = defineData({
  name: 'CoolName',
  schema,
  authorizationModes: {
    defaultAuthorizationMode: 'apiKey',
    apiKeyAuthorizationMode: {
      expiresInDays: 90,
    },
  },
})
```

Without name given (default):
```
AMPLIFY_DATA_GRAPHQL_ENDPOINT 
AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME
AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_KEY
```

With name given:
```
COOL_NAME_GRAPHQL_ENDPOINT 
COOL_NAME_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME
COOL_NAME_MODEL_INTROSPECTION_SCHEMA_KEY
```

In this case, in the `amplify/auth/post-confirmation/handler.ts`, this line `const { resourceConfig, libraryOptions } = await getAmplifyDataClientConfig(env)` fails to generate the `resourceConfig` because the `isDataClientEnv(env)` used in the `getAmplifyDataClientConfig()` always returns `false`.

Code below taken from [here](https://github.com/aws-amplify/amplify-backend/blame/560878f3af595d4a661f396545fe970fac39504d/packages/backend-function/src/runtime/get_amplify_clients_configuration.ts#L20) in the recent pull commit [2224](https://github.com/aws-amplify/amplify-backend/pull/2224).
```
const isDataClientEnv = (env: unknown): env is DataClientEnv => {
  return (
    env !== null &&
    typeof env === 'object' &&
    'AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME' in env && // <-- FAIL
    'AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_KEY' in env &&  // <-- FAIL
    'AWS_ACCESS_KEY_ID' in env &&
    'AWS_SECRET_ACCESS_KEY' in env &&
    'AWS_SESSION_TOKEN' in env &&
    'AWS_REGION' in env &&
    'AMPLIFY_DATA_GRAPHQL_ENDPOINT' in env && // <-- FAIL
    typeof env.AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME ===
      'string' && // <-- FAIL
    typeof env.AMPLIFY_DATA_MODEL_INTROSPECTION_SCHEMA_KEY === 'string' && // <-- FAIL
    typeof env.AWS_ACCESS_KEY_ID === 'string' &&
    typeof env.AWS_SECRET_ACCESS_KEY === 'string' &&
    typeof env.AWS_SESSION_TOKEN === 'string' &&
    typeof env.AWS_REGION === 'string' &&
    typeof env.AMPLIFY_DATA_GRAPHQL_ENDPOINT === 'string' // <-- FAIL
  )
}
```

### Reproduction steps

Follow the example in the [doc ](https://docs.amplify.aws/react/build-a-backend/functions/examples/create-user-profile-record/)with an additional step to add the name to `defineData()`.

1. Give the API a name in `defineData()`
```
// amplify/data/resource.ts
import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
import { postConfirmation } from "../auth/post-confirmation/resource";

const schema = a
  .schema({
    UserProfile: a
      .model({
        email: a.string(),
        profileOwner: a.string(),
      })
      .authorization((allow) => [
        allow.ownerDefinedIn("profileOwner"),
      ]),
  })
  .authorization((allow) => [allow.resource(postConfirmation)]);
export type Schema = ClientSchema<typeof schema>;

export const data = defineData({
  name: "CoolName", // Give the API a name
  schema,
  authorizationModes: {
    defaultAuthorizationMode: "apiKey",
    apiKeyAuthorizationMode: {
      expiresInDays: 30,
    },
  },
});
```
2. Create `post-confirmation` resource file `amplify/auth/post-confirmation/resource.ts`:
```
// amplify/auth/post-confirmation/resource.ts
import { defineFunction } from '@aws-amplify/backend';

export const postConfirmation = defineFunction({
  name: 'post-confirmation',
});
```
3. Create `post-confirmation` handler file `amplify/auth/post-confirmation/handler.ts`:
```
// amplify/auth/post-confirmation/handler.ts
import type { PostConfirmationTriggerHandler } from "aws-lambda";
import { type Schema } from "../../data/resource";
import { Amplify } from "aws-amplify";
import { generateClient } from "aws-amplify/data";
import { getAmplifyDataClientConfig } from '@aws-amplify/backend/function/runtime';
import { env } from "$amplify/env/post-confirmation";

const { resourceConfig, libraryOptions } = await getAmplifyDataClientConfig(
  env
);

Amplify.configure(resourceConfig, libraryOptions);

const client = generateClient<Schema>();

export const handler: PostConfirmationTriggerHandler = async (event) => {
  await client.models.UserProfile.create({
      email: event.request.userAttributes.email,
      profileOwner: `${event.request.userAttributes.sub}::${event.userName}`,
  });

  return event;
};
```
4. Set trigger in `amplify/auth/resource.ts`
```
// amplify/auth/resource.ts
import { defineAuth } from '@aws-amplify/backend';
import { postConfirmation } from './post-confirmation/resource';

export const auth = defineAuth({
  loginWith: {
    email: true,
  },
  triggers: {
    postConfirmation
  }
});
```

**Comment by stocaaro** [comment]
Hello @lengieng, Thank you for the detailed notes on this issue. I am working on a fix and will post updates as I make progress.

**Comment by stocaaro** [comment]
@lengieng, The fix for this was released with `@aws-amplify/backend@1.11.0`. Can you confirm that the feature is working for you with the latest version?

**Comment by dlamon1** [comment]
Not OP, I'm gettin the following error from 1.11.0

Argument of type 'ResourceConfig | { invalidType: "Some of the AWS environment variables needed to configure Amplify are missing."; }' is not assignable to parameter of type 'ResourcesConfig | LegacyConfig | AmplifyOutputs'.
  Type '{ invalidType: "Some of the AWS environment variables needed to configure Amplify are missing."; }' is not assignable to type 'ResourcesConfig | LegacyConfig | AmplifyOutputs'.ts(2345)
const resourceConfig: ResourceConfig | {
    invalidType: "Some of the AWS environment variables needed to configure Amplify are missing.";
}

`import { type Schema } from '../../data/resource';
import { Amplify } from 'aws-amplify';
import { generateClient } from 'aws-amplify/data';
import { getAmplifyDataClientConfig } from '@aws-amplify/backend/function/runtime';
import { env } from '$amplify/env/listTables';

const { resourceConfig, libraryOptions } = await getAmplifyDataClientConfig(env);

Amplify.configure(resourceConfig, libraryOptions);`

**Comment by stocaaro** [comment]
@dlamon1,

I just stepped through the [docs guide](https://docs.amplify.aws/react/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/) with a newly create react app. I get the same error you're seeing when I attempt to use `getAmplifyDataClientConfig` configuration when I haven't added permission to my data schema to allow it to be accessed by my function.

```ts
import { functionWithDataAccess } from '../functions/data-access/resource';
const schema = a.schema({
...
}).authorization((a) => [a.resource(functionWithDataAccess)]);
```

If you've done this and you're still seeing the error, can you share your `.amplify/generated/env/listTables.ts` file contents?

Thanks!

**Comment by dlamon1** [comment]
Sure, though what I posted is most of that resouce.ts

data/resource.ts

```
import { type ClientSchema, a, defineData } from '@aws-amplify/backend';
import { HardwareSchema } from './hardware.schema';
import { ProjectSchema } from './project.schema';
import { ConnectorSchema } from './connector.schema';
import { ManufacturerSchema } from './manufacturer.schema';
// import { FunctionsSchema } from './functions.schema';
import { onAdmin } from '../functions/onAdmin/resource';
import { listTables } from '../functions/listTables/resource';
import { clearRowsInTable } from '../functions/clearRowsInTable/resource';

const schema = a
  .schema({
    ...HardwareSchema,
    ...ProjectSchema,
    ...ConnectorSchema,
    ...ManufacturerSchema,
    // ...FunctionsSchema
    listTables: a
      .query()
      .arguments({})
      .returns(a.string())
      .handler(a.handler.function(listTables))
      .authorization((allow) => allow.group('ADMINS'))
  })
  .authorization((allow) => [allow.resource(listTables)]);

export type Schema = ClientSchema<typeof schema>;

export const data = defineData({
  schema,
  authorizationModes: {
    defaultAuthorizationMode: 'userPool'
  }
});
```

functions/listTables/resource.ts

```
// import { DynamoDB } from '@aws-sdk/client-dynamodb';
import type { Handler } from 'aws-lambda';
import { AmplifyClient, ListBranchesCommand } from '@aws-sdk/client-amplify';
import type { S3Handler } from 'aws-lambda';
import { type Schema } from '../../data/resource';
import { Amplify } from 'aws-amplify';
import { generateClient } from 'aws-amplify/data';
import { getAmplifyDataClientConfig } from '@aws-amplify/backend/function/runtime';
import { env } from '$amplify/env/listTables';

const { resourceConfig, libraryOptions } = await getAmplifyDataClientConfig(env);

Amplify.configure(resourceConfig, libraryOptions);

const client = generateClient<Schema>();

// Initialize the Amplify client
export const handler: Handler = async () => {}
```

**Comment by stocaaro** [comment]
@dlamon1 ,

Interesting. Your code from `functions/listTables/resource.ts` looks like the content I would expect to see in `functions/listTables/handler.ts`.

The function `resource.ts` files usually look something like:
```ts
import { defineFunction } from '@aws-amplify/backend';

export const listTables = defineFunction({
  name: 'listTables',
});
```

In `data/resource.ts` you have `import { listTables } from '../functions/listTables/resource';`, but `listTables` isn't exported in the copied code. I would guess that this is in the file, but outside the context you provided.

If you are merging the function resource and handler definition into a single file like this, you'll need to separate them out. The `resource.ts` files compose the CDK constructs needed to synthesize your backend where the `handler.ts` files are built and the built artifacts are deployed to lambda.

When I merge these files like this, I get other import error `Cannot find package '$amplify'`. I'm not sure how this would build to have the error you're encountering.

**Comment by dlamon1** [comment]
You're very kind at calling out my obvious mistake, thank you, you're right. 

```
import { defineFunction } from '@aws-amplify/backend';

export const listTables = defineFunction({
  name: 'listTables'
});
```

**Comment by dlamon1** [comment]
*To clarify, this was the the resource.ts file I had already implement when receiving this error.

**Comment by dlamon1** [comment]
The listTables.ts file

```
// This file is auto-generated by Amplify. Edits will be overwritten.
export const env = process.env as any;
```

**Comment by stocaaro** [comment]
@dlamon1, Looking at the content for `.amplify/generated/env/listTables.ts` helps. This is the content that gets written to the file right before the function is built. When the build succeeds the fully typed content is written. I haven't been able to replicate this in my example project. Would it be possible to provide an example project repo that is experiencing the issue? It would be helpful to get your `package.json`, 'tsconfig` and any lint config files,  but not as helpful as having a complete example of the issue.

**Comment by jserrano-alq** [comment]
I'm having this issue for a Angular project, its a fresh project following the [Create a user profile record
](https://docs.amplify.aws/angular/build-a-backend/functions/examples/create-user-profile-record/) 

data/resource.ts
```typescript
const schema = a.schema({
  NLBUser: a.model({
    id: a.string(),
    ...
    profileOwner: a.string(),
    ...
  }).authorization((allow) => [
    allow.ownerDefinedIn("profileOwner")
  ]),
...
}).authorization((allow) => [allow.publicApiKey(), allow.resource(postConfirmation)]);
```
auth/resource.ts
```typescript
import { defineAuth } from '@aws-amplify/backend';
import { postConfirmation } from './post-confirmation/resource';

/**
 * Define and configure your auth resource
 * @see https://docs.amplify.aws/gen2/build-a-backend/auth
 */
export const auth = defineAuth({
  loginWith: {
    email: true,
  },
  triggers: {
    postConfirmation
  }
});
```
auth/post-confirmation/resource.ts
```typescript
import { defineFunction } from '@aws-amplify/backend';

export const postConfirmation = defineFunction({
    name: 'post-confirmation',
});
```
auth/post-confirmation/handler.ts
```typescript
import type { PostConfirmationTriggerHandler } from "aws-lambda";
import { type Schema } from "../../data/resource";
import { Amplify } from "aws-amplify";
import { generateClient } from "aws-amplify/data";
import { getAmplifyDataClientConfig } from '@aws-amplify/backend-function/runtime';
import { env } from "$amplify/env/post-confirmation";

const { resourceConfig, libraryOptions } = await getAmplifyDataClientConfig(
  env
);

Amplify.configure(resourceConfig, libraryOptions);

const client = generateClient<Schema>();

export const handler: PostConfirmationTriggerHandler = async (event) => {
  await client.models.NLBUser.create({
    email: event.request.userAttributes.email,
    profileOwner `${event.request.userAttributes.sub}::${event.userName}`,
  });

  return event;
};

```

On the statement `Amplify.configure(resourceConfig, libraryOptions);` I get the error

```bash
Argument of type '{ invalidType: "Some of the AWS environment variables needed to configure Amplify are missing."; }' is not assignable to parameter of type 'ResourcesConfig | LegacyConfig | AmplifyOutputs'.
```

On the statement

```typescript
const { resourceConfig, libraryOptions } = await getAmplifyDataClientConfig(
  env
);
```
 Both `resourceConfig` and `resourceConfig` are returning
```
invalidType: "Some of the AWS environment variables needed to configure Amplify are missing.";
```

Where could I find those missing variables or how can i define them? Thanks!

**Comment by stocaaro** [comment]
Hi @jserrano-alq,

Thank you for all the details. I think I've gone through the same process, [starting an Angular app](https://v17.angular.io/guide/setup-local) and [setting up amplify](https://docs.amplify.aws/angular/start/manual-installation/) with an auth trigger to [create a profile record](https://docs.amplify.aws/angular/build-a-backend/functions/examples/create-user-profile-record/).

I didn't encounter the error you're seeing, but I've [capture my example app here](https://github.com/stocaaro/amplify-backend-issue-2320-angular).

Locally, I'm using node v20.18.0 (npm v10.8.2), which versions are you building with?

You're right that using `'../amplify_outputs.json'` directly doesn't work. Both sandbox and deploy don't construct the outputs file until after the cdk constructs have been synthesized. Locally, the sandbox will use the outputs file built by a prior run, but deploying only runs once, so will always encounter that error.

I don't see any obvious mistakes or differences and I appreciate your help in working to track down the difference that is causing this error in your environment.

**Comment by spyderbr6** [comment]
ugh my whole morning was thrown away trying to troubleshoot this. i'm glad there was a patch but now i'm in the same place jseranno-alq is after the patch... i'll try to troubleshoot more tonight and will see if i can leverage claude. will post findings if something obvious comes out of it.

**Comment by dryhurst** [comment]
@spyderbr6 every day of the last 4 months has been thrown away dealing with how horrible aws-amplify is, it's chock full of issues, bugs, broken features, poor documentation. it's a literal disaster and i regret my clients decision to use it as a stack, but now we're committed and dealing with yet another bs issue as described in this ticket. aws-amplify has cost my business thousands of dollars in lost revenue.

**Comment by dryhurst** [comment]
in my case .env files are not being populated at all and therefore i get the above error previously reported `Argument of type '{ invalidType: "Some of the AWS environment variables needed to configure Amplify are missing."; }' snipped`

**Comment by stocaaro** [comment]
Hello @dryhurst and @spyderbr6,

I understand how frustrating it is when things don't work right. If you can provide either an example app repo that exhibits this problem or step by step instructions (including cli commands used and node version) to reproduce the problem, it would help accelerate our investigation. I've tried several approaches without running into the issue and will be investigating further today.

Prior to releasing `getAmplifyDataClientConfig` the recommended way to make api calls inside a function was to use the `client.graphql` to call [queries generated by codegen](https://docs.amplify.aws/react/deploy-and-host/sandbox-environments/features/#generate-client-codegen).

<details>

<summary>
If your blocked on this issue and are looking for an workaround to get unblocked while we investigate, here are the docs for postConfirmation that use client.graphql.
</summary>


Run the command `npx ampx sandbox` to create the backend, then use the command below to generate GraphQL client code to call your data backend.

```sh title="Terminal" showLineNumbers={false}
npx ampx generate graphql-client-code --out <path-to-post-confirmation-handler-dir>/graphql
```

Then, create the corresponding handler file, `amplify/auth/post-confirmation/handler.ts`, file with the following contents:

```ts title="amplify/auth/post-confirmation/handler.ts"
import type { PostConfirmationTriggerHandler } from "aws-lambda";
import { type Schema } from "../../data/resource";
import { Amplify } from "aws-amplify";
import { generateClient } from "aws-amplify/data";
import { env } from "$amplify/env/post-confirmation";
import { createUserProfile } from "./graphql/mutations";

Amplify.configure(
  {
    API: {
      GraphQL: {
        endpoint: env.AMPLIFY_DATA_GRAPHQL_ENDPOINT,
        region: env.AWS_REGION,
        defaultAuthMode: "iam",
      },
    },
  },
  {
    Auth: {
      credentialsProvider: {
        getCredentialsAndIdentityId: async () => ({
          credentials: {
            accessKeyId: env.AWS_ACCESS_KEY_ID,
            secretAccessKey: env.AWS_SECRET_ACCESS_KEY,
            sessionToken: env.AWS_SESSION_TOKEN,
          },
        }),
        clearCredentialsAndIdentityId: () => {
          /* noop */
        },
      },
    },
  }
);

const client = generateClient<Schema>({
  authMode: "iam",
});

export const handler: PostConfirmationTriggerHandler = async (event) => {
  await client.graphql({
    query: createUserProfile,
    variables: {
      input: {
        email: event.request.userAttributes.email,
        profileOwner: `${event.request.userAttributes.sub}::${event.userName}`,
      },
    },
  });

  return event;
};

```

The full diff from the related docs update can be found [here](https://github.com/aws-amplify/docs/pull/8096/files).
</details>

**Comment by jserrano-alq** [comment]
Thank you so much @stocaaro ! I confirm this work around works just fine.

**Comment by spyderbr6** [comment]
i may have just needed to trigger a rebuild on my sandbox because (after the node packages were updated to this latest patch) vscode kept saying that error and so i just put logs in place in the handler before and after the retrieval to see what was there and miraculously it started working. thanks guys for the patch.

**Comment by spyderbr6** [comment]
there is something wonky with sandbox i think. i can't put my finger on it yet. at this point i have a stage branch that properly generates all the keys in .amplify/generated/env . i've branched from it and added a new function and now they all fail because amplify isnt generating any keys for anything in the new branch.

**Comment by spyderbr6** [comment]
i've been able to consistently reproduce this. while sandbox is running, i add an empty folder. it attempts to rebuild and starts failing everything. 
[amplifyfailonfunction.txt](https://github.com/user-attachments/files/18296193/amplifyfailonfunction.txt)
here^

removing the folder yields an error but it appears to regenerate the files. 

Error
        Affected versions: bootstrap: <21
        More information at: https://github.com/aws/aws-cdk/issues/31885

If you don’t want to see a notice anymore, use "cdk acknowledge <id>". For example, "cdk acknowledge 31885".
Lambda bundled into an empty zip
Caused By: ❌  amplify-amplifyvitereacttemplate-Desktop-sandbox-acfc7abd3b failed: InvalidParameterValueException: Uploaded file must be a non-empty zip

Resolution: Try removing '.amplify/artifacts' then running the command again. If it still doesn't work, see https://github.com/aws/aws-cdk/issues/18459 for more methods.

[Sandbox] Watching for file changes...

**Comment by spyderbr6** [comment]
i'm done mucking with it for the night. basically if anything is wrong with any function anywhere it kills all the generated environment files and nothing will work. basically replaces them with placeholders. makes it hard to troubleshoot because you cant get to a real error to see whats wrong with the function.

**Comment by spyderbr6** [comment]
ok. i'm closing out my engagement on this one. i think i'm in a good spot. 
1. turn off sandbox. 
2. manually reinstalled all the packages. npm was out of sync where it thought i had a package but i didnt. probably due to branch switching. outdated and such commands didnt matter, manually reinstall all the aws packages.
3. wipe out the cdk artifacts. the whole folder including the env folder.
4. ensure the function is completely error-free. i didnt negative test this after the first 3 steps so idk if it will still perform the wonkiness at this point. seems like the backend build process maybe could always generate environments first then attempt to run, which could enable better error checking rather than failing on step 1 when step 8 was really the problem.
5. goodluck if anyone else runs into this.

**Comment by stocaaro** [comment]
@spyderbr6 ,

Glad you have a process that is working and I really appreciate your notes. If I'm reading correctly, the steps you're following are roughly in line with the "Resolution:" guidance from the sandbox error output you posted above. Do you have any recommendations on how we might improve upon this guidance?

A couple questions for anyone encountering this,
- Are there any cases where you are seeing the type issue that don't correspond to an error captured in the sandbox output with a "Resolution:" suggestion?
- Would a change to the type error that you get when the env variables are missing be more clear if it specifically directed you to review the sandbox output for function failures?

**Comment by spyderbr6** [comment]
it would be better to get an error on why the function is failing. amplify fails on non related functions and the errors (at least from my untrained eye) are only related to the fact that the aws environment variables arent in the file. its reporting on the symptom, not why or where the process as a whole is failing. in my instance i have a push notification handler THATS where the issue is occurring but the error shown is related to my process cleanup function which has nothing to do with it. the unrelated function error is because amplify is failing to generate files across all functions and just reports on the first failure.

**Comment by spyderbr6** [comment]
it would also be helpful if it cleaned itself up. all the cached cdk files have to be wiped when an error occurs. its a workaround which i'm now trained to do but its what triggered me to reply here.

**Comment by stocaaro** [comment]
Talking with the team, the pattern of raising typescript errors was overly confusing and I've adjusted `getAmplifyDataClientConfig` to have specific type expectations on env variables. In testing, this experience should be better. The team is looking at ways to improve upon build failures.

@spyderbr6 , If you would like to advocate for the clear and retry approach you describe above as a [feature request](https://github.com/aws-amplify/amplify-backend/issues/new?assignees=&labels=pending-triage&projects=&template=feature-request.yml), could you capture this in a [new issue](https://github.com/aws-amplify/amplify-backend/issues/new?assignees=&labels=pending-triage&projects=&template=feature-request.yml)?

The original issue has been fixed and a change has been merged to better manage `env` input types. When this change is released, I plan to close this issue unless it is clear that there are remaining problems related to the `getAmplifyDataClientConfig` behavior.

Thanks!
@stocaaro

**Comment by lengieng** [comment]
@stocaaro My apologies for the late response. I confirm the `@aws-amplify/backend@1.11.0` and later are working for me.

Thanks so much for the fix!
@lengieng
